Code/Framework/AzFramework/AzFramework/Input/Channels/InputChannelId.h

```diff
@@ -22,8 +23,7 @@ namespace AzFramework
     public:
         ////////////////////////////////////////////////////////////////////////////////////////////
         // Constants
-        static const int NAME_BUFFER_SIZE = 64;
-        static const int MAX_NAME_LENGTH = NAME_BUFFER_SIZE - 1;
+        static const int MAX_NAME_LENGTH = 63;
```

Comment:
This could be made constexpr as well.

Also I would have stuck with 64.

If the goal was to fit an InputChannelId into a 64 byte cache line, the fixed_string would have to be 58 characters.

This is due to the fact that this class has a 4 byte Crc32 member, leaving 60 bytes to play with it.
The fixed_string class only uses 1 byte to maintain the size when the capacity is 255 characters or less.
So there is 59 bytes left. The fixed_string class always stores has 1 additional byte than the `NumElements` argument for the null-terminator.
So `fixed_string<58>` would have to be used to fit it within a cache line.

Since that is not occurring, I see it as better to use a `fixed_string<64>`.
Since it is a power of 2 it is familiar to most developers using it and more importantly it reduces the number of total template instantiations, since there are `fixed_string<64>`  instances being used in other places.

Reply:
I've updated this to use a `fixed_string<64>` member.

Reply:
My initial suggestion to use 63 was only to keep the prior behavior the same, but I agree that 64 is a better choice.